app.js

```diff
@@ -1,8 +1,84 @@
-function hello(name) {
-  const greeting = `hello ${name}`;
-  return greeting;
+const _ = require('lodash');
+
+const SFDX = require('./sfdx');
+
+async function deployProject(action, settings) {
+  const {
+    consumerKey,
+    jwtKey,
+    username,
+    sourceDirectory,
+    instanceUrl,
+    testLevel
+  } = mergeInputs(action.params, settings);
+
+  await SFDX.authenticate({
+    consumerKey,
+    jwtKey,
+    username,
+    sourceDirectory
+  });
+
+  return SFDX.deployProject({
+    sourceDirectory,
+    instanceUrl,
+    testLevel,
+    username
+  });
+}
+
+async function validateProject(action, settings) {
+  const {
+    username,
+    consumerKey,
+    jwtKey,
+    sourceDirectory,
+    instanceUrl,
+    testLevel
+  } = mergeInputs(action.params, settings);
+
+  await SFDX.authenticate({
+    consumerKey,
+    jwtKey,
+    username,
+    sourceDirectory
+  });
+
+  return SFDX.validateProject({
+    sourceDirectory,
+    instanceUrl,
+    testLevel,
+    username
+  });
+}
+
+function mergeInputs(params, settings) {
```

Comment:
It's not really domain related function, I'd consider moving it to separate `helpers.js` file

Reply:
The code seemed too small to separate it, but you'r "domain" argument convinces me.